### ISSUE TITLE
IntelliJ IDEA 2017.2 NPE

### ISSUE THREAD

**Issue by ddska** [hsz/idea-gitignore#389]
```
null
java.lang.NullPointerException
	at mobi.hsz.idea.gitignore.util.CacheMap.isFileIgnored(CacheMap.java:290)
	at mobi.hsz.idea.gitignore.IgnoreManager.isFileIgnored(IgnoreManager.java:321)
	at mobi.hsz.idea.gitignore.vcs.IgnoreFileStatusProvider.getFileStatus(IgnoreFileStatusProvider.java:69)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.calcStatus(FileStatusManagerImpl.java:135)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.getStatus(FileStatusManagerImpl.java:248)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.b(LineStatusTrackerManager.java:271)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.b(LineStatusTrackerManager.java:235)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.b(LineStatusTrackerManager.java:215)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.access$1000(LineStatusTrackerManager.java:56)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager$MyFileStatusListener.fileStatusesChanged(LineStatusTrackerManager.java:416)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.fileStatusesChanged(FileStatusManagerImpl.java:194)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.b(FileStatusManagerImpl.java:186)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:314)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:821)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:649)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by Passto** [comment]
I have same problem. After update to the latest version of Intellij IDEA. 

Same error message and stack trace.

Environment:
_IntelliJ IDEA 2017.2
Build #IU-172.3317.76, built on July 15, 2017
JRE: 1.8.0_152-release-915-b5 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 7 6.1_

**Comment by hsz** [comment]
@ddska @Passto Thanks, it's fixed out of the box with the forthcomming v2.0.0 because of migration from custom CacheMap component to native IDE indexing.

Ref: #367
